src/plugin/sdl/sdl.c

```diff
@@ -1019,6 +1179,182 @@ static void sdl_scrub(void)
   }
 }
 
+#if defined(HAVE_SDL2_TTF) && defined(HAVE_FONTCONFIG)
+static void SDL_draw_string(void *opaque, int x, int y, unsigned char *text, int len, Bit8u attr)
+{
+  char *s;
+  struct char_set_state state;
+  int characters;
+  t_unicode *str;
+
+  v_printf("SDL_draw_string\n");
+
+  init_charset_state(&state, trconfig.video_mem_charset);
+  characters = character_count(&state, (char *)text, len);
+  if (characters == -1) {
+    v_printf("SDL: invalid char count\n");
+    return;
+  }
+  str = malloc(sizeof(t_unicode) * (characters + 1));
+
+  charset_to_unicode_string(&state, str, (const char **)&text, len, characters + 1);
+  cleanup_charset_state(&state);
+
+  s = unicode_string_to_charset((wchar_t *)str, "utf8");
+  free(str);
+  pthread_mutex_lock(&sdl_font_mtx);
+  if (!sdl_font)
+    v_printf("SDL: sdl_font is null\n");
+
+  SDL_Surface *srf = TTF_RenderUTF8_Shaded(sdl_font, s,
+                                           text_colors[ATTR_FG(attr)],
+                                           text_colors[ATTR_BG(attr)]);
+  pthread_mutex_unlock(&sdl_font_mtx);
+  free(s);
+  SDL_Texture *txt = SDL_CreateTextureFromSurface(renderer, srf);
```

Comment:
No-no, this whole is quite wrong. :)
Please make text_lock() to do SDL_LockSurface()
to lock our main surface.
Then here just use SDL_BlitSurface().

Reply:
okay, I'll investigate. After I solved the other problems you just raised.

Reply:
OK, that was a very silly request
from me. Now that I played with your
code, I understand it shouldn't (and
cannot) be done that way.
Your rendering code was correct.
Sorry for wasting your time.

I gave your code a re-spin in #1398.
So we can move the review/discussions there.